### ISSUE TITLE
Extend Language Services with custom rules

### ISSUE THREAD

**Issue by mohammedzamakhan** [angular/angular#28528]
# 🚀 feature request

### Relevant Package
@angular/language-services

### Description
I wanted to extend language services to give me linting issues related to accessibility as well, like suppose if the label doesn't have a for attribute or input doesn't have id attribute etc to show a warning in the HTML template.

### Describe the solution you'd like
Ability to extend and add rules to the ngc, so that we can add custom linting rules to the project

### Describe alternatives you've considered
I was considering to use htmlhint as well, but was thinking extending ngc can give more capability.

**Comment by alxhub** [comment]
I think this is out of scope for Angular - if you want to write your own linter/integration into IDEs for accessibility checks, feel free. It's hard for us to support a custom plugin interface, though, as we'd have to maintain compatibility as we evolved the Language Service.
